ksqldb-functional-tests/src/test/resources/rest-query-validation-tests/pull-queries-against-ctas-materialized.json

```diff
@@ -205,6 +205,613 @@
         "message": "The `USERS` table isn't queryable. To derive a queryable table, you can do 'CREATE TABLE QUERYABLE_USERS AS SELECT * FROM USERS'. See https://cnfl.io/queries for more info.",
         "status": 400
       }
+    },
+    {
+      "name": "persistent query with KEY filter and projection +++ pull query table scan and single key lookup",
+      "statements": [
+        "CREATE TABLE INPUT (ID STRING PRIMARY KEY, GRADE STRING, RANK INT) WITH (kafka_topic='test_topic', value_format='JSON');",
+        "CREATE TABLE NEW_INPUT AS SELECT ID, GRADE, RANK * 2 AS DOUBLE_RANK FROM INPUT WHERE ID='11';",
+        "SELECT * FROM NEW_INPUT;",
+        "SELECT ID, DOUBLE_RANK FROM NEW_INPUT;",
+        "SELECT * FROM NEW_INPUT WHERE ID='11';",
+        "SELECT ID, DOUBLE_RANK FROM NEW_INPUT WHERE ID='11';",
+        "SELECT * FROM NEW_INPUT WHERE ID='12';",
+        "SELECT ID, DOUBLE_RANK FROM NEW_INPUT WHERE ID='12';"
+      ],
+      "inputs": [
+        {"topic": "test_topic", "timestamp": 12346, "key": "11", "value": {"GRADE": "A", "RANK": 1}},
+        {"topic": "test_topic", "timestamp": 12345, "key": "10", "value": {"GRADE": "B", "RANK": 2}},
+        {"topic": "test_topic", "timestamp": 12389, "key": "12", "value": {"GRADE": "C", "RANK": 3}}
+      ],
+      "responses": [
+        {"admin": {"@type": "currentStatus"}},
+        {"admin": {"@type": "currentStatus"}},
+        {"query": [
+          {"header":{"schema":"`ID` STRING KEY, `GRADE` STRING, `DOUBLE_RANK` INTEGER"}},
+          {"row":{"columns":["11", "A", 2]}}
+        ]},
+        {"query": [
+          {"header":{"schema":"`ID` STRING KEY, `DOUBLE_RANK` INTEGER"}},
+          {"row":{"columns":["11", 2]}}
+        ]},
+        {"query": [
+          {"header":{"schema":"`ID` STRING KEY, `GRADE` STRING, `DOUBLE_RANK` INTEGER"}},
+          {"row":{"columns":["11", "A", 2]}}
+        ]},
+        {"query": [
+          {"header":{"schema":"`ID` STRING KEY, `DOUBLE_RANK` INTEGER"}},
+          {"row":{"columns":["11", 2]}}
+        ]},
+        {"query": [
+          {"header":{"schema":"`ID` STRING KEY, `GRADE` STRING, `DOUBLE_RANK` INTEGER"}}
+        ]},
+        {"query": [
+          {"header":{"schema":"`ID` STRING KEY, `DOUBLE_RANK` INTEGER"}}
+        ]}
+      ]
+    },
+    {
+      "name": "persistent query with KEY filter and projection +++ pull query table scan and single key lookup ***FAILURE***",
```

Comment:
nit: should we explicitly state the the error root cause here? Since otherwise the name is exactly the same as above except we say it is a `***FAILURE***` case.